src/main/java/duke/Task.java

```diff
@@ -0,0 +1,120 @@
+/**
+ * The Task object contains parameters and information which defines the Task, such as the name, time, and whether it
+ * is done. This object also handles any editing to the contents of the Task.
+ */
+
+package duke;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+abstract class Task {
+    private char taskType;
+    private boolean isDone;
+    private String taskName;
+    private LocalDateTime taskTime;
+
+    /**
+     * Constructor for a new generic Task object.
+     * @param taskType defines the type of this Task.
+     * @param isDone whether the Task is done.
+     * @param taskName the name of the Task.
+     * @param taskTime the end time of the Task.
+     */
+    public Task(char taskType, boolean isDone, String taskName, String taskTime) {
+        this.taskType = taskType;
+        this.isDone = isDone;
+        this.taskName = taskName;
+        this.taskTime = taskTime.isEmpty()
+                ? LocalDateTime.MIN
+                : this.parseDate(taskTime);
+    }
+
+    /**
+     * Parses the date and time in a string passed into the constructor.
+     * @param taskTime the end time of this Task.
+     * @return a LocalDateTime object containing a string representation of a time.
+     *         If no time is present, a default time "2359" is used.
+     * @throws DateTimeParseException if the string is not in d/M/yyyy HHmm format.
+     */
+    public LocalDateTime parseDate(String taskTime) throws DateTimeParseException {
+        if (taskType == 'D' && !taskTime.contains(" ")) {
+            taskTime = taskTime + " 2359";
+        }
+        return LocalDateTime.parse(taskTime, DateTimeFormatter.ofPattern("d/M/yyyy HHmm"));
+    }
+
+    /**
+     * Returns if this Task is done.
+     * @return the status of the Task.
+     */
+    public boolean getStatus() {
```

Comment:
Perhaps this is better named as `getDoneStatus()`, or `isDone()`? I feel that status refers to something more generic than a done/not done flag.